graphium/ipu/ipu_losses.py

```diff
@@ -29,7 +30,8 @@ def forward(self, input: Tensor, target: Tensor) -> Tensor:
 
         # Replace the nan-targets by 0 or 1. Take the value closest to the input.
         # Give a weight of 0 where there are nan-targets
-        nan_targets = target.isnan()
+        #nan_targets = target.isnan()
```

Comment:
Although the name of the class ends with `IPU`, this class might be used on CPU or GPU, for reproducibility, debugging, or laziness. So I would recommend creating a new function 

```python
from graphium.ipu import is_running_on_ipu
def device_is_nan(tensor):
    if is_running_on_ipu():
        return ipu_isnan(tensor)
    else:
        return torch.isnan(tensor)


```